### ISSUE TITLE
Support code_point.range for any byte encoding

### ISSUE THREAD

**Issue by Spartan322** [foonathan/lexy#168]
Perhaps I'm missing a manner to do this otherwise, but while not strictly required as it can be done with a lot of tedious work, it would be really nice to have a quick manner to reference a range within any encoding and not just unicode code points.

**Comment by foonathan** [comment]
To clarify, you want something like `dsl::byte_range<0x10, 0x20>` which skips a byte in the range `[0x10, 0x20]`? And you specifically want something for byte encodings, not texts?

**Comment by Spartan322** [comment]
Yeah if I had something that could evaluate in such a way it would be a lot more convenient for my use cases.

**Comment by foonathan** [comment]
I've added `dsl::byte.range<0x10, 0x20>()` and `dsl::bytes<N>.range<0x10, 0x20>()` (as well as other predicates).
